Fails to import with numpy 1.23.5 after compiling on raspbian arm with clang 13  for python 3.11

### Steps to reproduce:

Pip install build from source as there is no wheel for python 3.11

```
(homeassistant-shadow) homeassistant@lazaro:/srv/homeassistant-shadow$ pip install numpy
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting numpy
  Downloading numpy-1.23.5.tar.gz (10.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.7/10.7 MB 8.4 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: numpy
  Building wheel for numpy (pyproject.toml) ... done
  Created wheel for numpy: filename=numpy-1.23.5-cp311-cp311-linux_armv7l.whl size=10681800 sha256=38fff5fe9473a06d14419d01d7e1321906d59765930bd20be43ce30f42e282b6
  Stored in directory: /home/homeassistant/.cache/pip/wheels/32/6f/c1/c2fda686f19bc32d64822ae92e3dc4922f9a999c703d998abc
Successfully built numpy
Installing collected packages: numpy
Successfully installed numpy-1.23.5
```

### Error message:

```shell
Traceback (most recent call last):
  File "/srv/homeassistant-shadow/lib/python3.11/site-packages/numpy/core/__init__.py", line 23, in <module>
    from . import multiarray
  File "/srv/homeassistant-shadow/lib/python3.11/site-packages/numpy/core/multiarray.py", line 10, in <module>
    from . import overrides
  File "/srv/homeassistant-shadow/lib/python3.11/site-packages/numpy/core/overrides.py", line 6, in <module>
    from numpy.core._multiarray_umath import (
ImportError: /srv/homeassistant-shadow/lib/python3.11/site-packages/numpy/core/_multiarray_umath.cpython-311-arm-linux-gnueabihf.so: undefined symbol: __mulodi4

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/srv/homeassistant-shadow/lib/python3.11/site-packages/numpy/__init__.py", line 140, in <module>
    from . import core
  File "/srv/homeassistant-shadow/lib/python3.11/site-packages/numpy/core/__init__.py", line 49, in <module>
    raise ImportError(msg)
ImportError:

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.11 from "/srv/homeassistant-shadow/bin/python"
  * The NumPy version is: "1.23.5"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: /srv/homeassistant-shadow/lib/python3.11/site-packages/numpy/core/_multiarray_umath.cpython-311-arm-linux-gnueabihf.so: undefined symbol: __mulodi4
```


### Additional information:

Both python and numpy where compiled on the raspberry pi 4 running bullseye using clang-13

In particular, for compiling python the following was run

```
export CC=/usr/bin/clang-13
export CXX=/usrbin/clang++-13
export LDFLAGS=" -fuse-ld=lld"
ln -s /usr/bin/llvm-arm-13 /usr/local/bin/llv-arm
ln -s /usr/bin/llvm-profdata-13 /usr/local/bin/llvm-profdata

./configure --prefix /usr/local/cpython-3.11 --with-lto=full --enable-optimizations 
```

python -m test passes correctly.

The error showed up inside a venv created with the previously compiled and installed python 3.11.

I'm running another build to get the build log and will attach it when it's done.

Thanks for the report @jtoledo1974.

A few things that stand out in the build log. Lots of warnings like:
```
INFO: clang-13: numpy/core/src/multiarray/array_coercion.c
warning: overriding currently unsupported use of floating point exceptions on this target [-Wunsupported-floating-point-opt]
```

Feature testing for NEON failed - but that is probably expected and shouldn't matter, it just disables instruction sets:
```
2022-12-16T19:17:41,020   /tmp/pip-install-3wenzh90/numpy_76e85714f9594a2385101dc0adfea813/numpy/distutils/checks/cpu_neon_fp16.c:9:24: warning: implicit declaration of function 'vcvt_f32_f16' is invalid in C99 [-Wimplicit-function-declaration]
2022-12-16T19:17:41,022       float32x4_t v_z4 = vcvt_f32_f16((float16x4_t)vld1_s16(src));
2022-12-16T19:17:41,024                          ^
2022-12-16T19:17:41,025   /tmp/pip-install-3wenzh90/numpy_76e85714f9594a2385101dc0adfea813/numpy/distutils/checks/cpu_neon_fp16.c:9:17: error: initializing 'float32x4_t' (vector of 4 'float32_t' values) with an expression of incompatible type 'int'
2022-12-16T19:17:41,027       float32x4_t v_z4 = vcvt_f32_f16((float16x4_t)vld1_s16(src));
2022-12-16T19:17:41,029                   ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2022-12-16T19:17:41,030   2 warnings and 1 error generated.

2022-12-16T19:17:41,035   WARN: CCompilerOpt.feature_test[1563] : testing failed
```

And now that I've read the initial traceback correctly, the problem is:
```
/numpy/core/_multiarray_umath.cpython-311-arm-linux-gnueabihf.so: undefined symbol: __mulodi4
```
That seems to be a known bug in Clang, see e.g. https://codereview.chromium.org/2011073002

--- a/numpy/core/meson.build
+++ b/numpy/core/meson.build
@@ -315,7 +315,8 @@ optional_intrinsics = [
   ['__builtin_bswap32', '5u', [], []],
   ['__builtin_bswap64', '5u', [], []],
   ['__builtin_expect', '5, 0', [], []],
-  ['__builtin_mul_overflow', '5, 5, (int*)5', [], []],
+  # Test `long long` for arm+clang 13 (gh-22811, but we use all versions):
+  ['__builtin_mul_overflow', '(long long)5, 5, (int*)5', [], []],
 ]
 if host_machine.cpu_family() in ['x86', 'x86_64']
  optional_intrinsics += [

--- a/numpy/core/setup_common.py
+++ b/numpy/core/setup_common.py
@@ -129,7 +129,7 @@ MANDATORY_FUNCS = [
     "floor", "ceil", "sqrt", "log10", "log", "exp", "asin",
     "acos", "atan", "fmod", 'modf', 'frexp', 'ldexp',
     "expm1", "log1p", "acosh", "asinh", "atanh",
-    "rint", "trunc", "exp2", 
+    "rint", "trunc", "exp2",
     "copysign", "nextafter", "strtoll", "strtoull", "cbrt",
     "log2", "pow", "hypot", "atan2",
     "csin", "csinh", "ccos", "ccosh", "ctan", "ctanh",
@@ -178,7 +178,9 @@ OPTIONAL_INTRINSICS = [("__builtin_isnan", '5.'),
                        ("__builtin_bswap32", '5u'),
                        ("__builtin_bswap64", '5u'),
                        ("__builtin_expect", '5, 0'),
-                       ("__builtin_mul_overflow", '5, 5, (int*)5'),
+                       # Test `long long` for arm+clang 13 (gh-22811,
+                       # but we use all versions of __builtin_mul_overflow):
+                       ("__builtin_mul_overflow", '(long long)5, 5, (int*)5'),
                        # MMX only needed for icc, but some clangs don't have it
                        ("_m_from_int64", '0', "emmintrin.h"),
                        ("_mm_load_ps", '(float*)0', "xmmintrin.h"),  # SSE

